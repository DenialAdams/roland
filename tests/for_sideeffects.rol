proc start() -> u64 {
   print("Start expr");
   return 0;
}
proc end() -> u64 {
   print("End expr");
   return 3;
}

proc main() {
   for i in start()..end() {
      print("Loop body");
   }
}
