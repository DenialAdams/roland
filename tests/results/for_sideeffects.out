Start expr
End expr
Loop body
Loop body
Loop body
